src/preferArrayLiteralRule.ts

```diff
@@ -47,18 +70,40 @@ export class Rule extends Lint.Rules.AbstractRule {
 
         ruleOptions.forEach((opt: unknown) => {
             if (isObject(opt)) {
-                value = opt['allow-type-parameters'] === true;
+                allowSingleArgument = opt['allow-single-argument'] === true;
```

Comment:
Is this option name OK?

Reply:
Hmm, if the purpose of the argument is to allow sized creation, how about `allow-size-argument`? `single` might be interpreted as implying a more broad set of allowed behavior.

Reply:
Yep, this one makes it more clear.